### PR TITLE
include SysError, remove Error docs in Makefile

### PR DESCRIPTION
**Pull request by psahabu**
Addresses shortcoming in #5162 pointed out by @bradcray.

### CODE REVIEW

File: modules/standard/SysError.chpl [hunk 1]
```diff
@@ -19,21 +19,23 @@
...
+   an error message. This type can be returned from routines generating an
+   error.
 
    The IO module uses these routines in a way that supports error inspection
    and also rapid prototyping. Most routines in the IO module have two forms.
    In one form, an error (of type syserr) is returned in an out error argument.
    In the second form, no error is returned, and instead the task will halt
    with a fatal error if an error is encountered.
 
+   This module is unrelated to the implementation of error handling in Chapel
```
Comment (ben-albrecht) [inline modules/standard/SysError.chpl]: I would put this in a `.. note:`

File: modules/standard/SysError.chpl [hunk 2]
```diff
@@ -19,21 +19,27 @@
 
 /*
 
-   Support for error handling.
+   This module helps handle system errors.
 
-   This module helps to handle errors. In particular, it enables routines
-   to return a syserr - encoding an error state - and then contains routines
-   that can be provided a syserr in order to print out a useful error message.
+   In particular, it enables routines to return a ``syserr`` - encoding an
```
Comment (ben-albrecht) [inline modules/standard/SysError.chpl]: I might merge this sentence with the next few, starting with:

> This module defines the type ``syserr``, which can encode an error code or an error message. ...

**Comment by bradcray** [comment]
As a complete outsider to this...  I didn't mean for @psahabu to have to be responsible for maintenance of and improvements to pre-existing documentation.  I.e., isn't this the old documentation for Error.chpl simply renamed?

**Comment by psahabu** [comment]
@bradcray The wording of the existing documentation made less sense given the new error handling system, so it made sense for me to update and streamline it.

**Pull request merged**